sdk/src/main/java/com/kin/ecosystem/Kin.java

```diff
@@ -109,89 +198,68 @@ private static SignInData getJwtSignInData(@NonNull final String jwt) {
-	}
-
-	private static void initBlockchain(@NonNull final Context context) throws BlockchainException {
-		final String networkUrl = Configuration.getEnvironment().getBlockchainNetworkUrl();
-		final String networkId = Configuration.getEnvironment().getBlockchainPassphrase();
-		KinClient kinClient = new KinClient(context, new ServiceProvider(networkUrl, networkId) {
-			@Override
-			protected String getIssuerAccountId() {
-				return Configuration.getEnvironment().getIssuer();
-			}
-		}, KIN_ECOSYSTEM_STORE_PREFIX_KEY);
-		BlockchainSourceImpl.init(instance.eventLogger, kinClient, BlockchainSourceLocal.getInstance(context));
-	}
-
-	private static void initAuthRepository(@NonNull final Context context, @NonNull final SignInData signInData)
-		throws ClientException {
-		AuthRepository.init(AuthLocalData.getInstance(context),
-			AuthRemoteData.getInstance(instance.executorsUtil));
 		String deviceID = AuthRepository.getInstance().getDeviceID();
 		signInData.setDeviceId(deviceID != null ? deviceID : UUID.randomUUID().toString());
 		signInData.setWalletAddress(getPublicAddress());
 		AuthRepository.getInstance().setSignInData(signInData);
-	}
 
-	private static void initOfferRepository() {
-		OfferRepository.init(OfferRemoteData.getInstance(instance.executorsUtil), OrderRepository.getInstance());
-	}
+		ObservableData<String> observableData = AuthRepository.getInstance().getAppID();
+		String appID = observableData.getValue();
+		if (appID == null) {
+			observableData.addObserver(new Observer<String>() {
+				@Override
+				public void onChanged(String appID) {
+					BlockchainSourceImpl.getInstance().setAppID(appID);
+				}
+			});
+		}
+		BlockchainSourceImpl.getInstance().setAppID(appID);
 
-	private static void initOrderRepository(@NonNull final Context context) {
-		OrderRepository.init(BlockchainSourceImpl.getInstance(),
-			instance.eventLogger,
-			OrderRemoteData.getInstance(instance.executorsUtil),
-			OrderLocalData.getInstance(context, instance.executorsUtil));
+		if (!AccountManagerImpl.getInstance().isAccountCreated()) {
+			AccountManagerImpl.getInstance().start();
+			AccountManagerImpl.getInstance().addAccountStateObserver(new Observer<Integer>() {
+				@Override
+				public void onChanged(Integer value) {
+					switch (value) {
+						case AccountManager.PENDING_CREATION:
+							isAccountLoggedIn = true;
+							break;
+						case AccountManager.CREATION_COMPLETED:
+							instance.executorsUtil.mainThread().execute(new Runnable() {
+								@Override
+								public void run() {
+									loginCallback.onResponse(null);
+								}
+							});
+							break;
+						case ERROR:
+							instance.executorsUtil.mainThread().execute(new Runnable() {
+								@Override
+								public void run() {
+									loginCallback.onFailure(AccountManagerImpl.getInstance().getError());
+								}
+							});
+							break;
+					}
+				}
+			});
+		} else {
+			isAccountLoggedIn = true;
```

Comment:
what happens in case that account is already created? does the `addAccountStateObserver` will trigger  `loginCallback.onResponse`? it seems that it will never be called in this case? (cause you will not call `AccountManagerImpl.getInstance().start();`)